index.js

```diff
@@ -22,7 +22,20 @@ function extract (str, fn, options) {
   }
   // default filename is a javascript file (for backwards compatibility)
   var filename = (options && options.filename) || 'abc.js'
-  var regexp = cp.regex(filename)
+  var regexp = "";
+
+  // check for custom comment pattern
+  // for example support for assembler files may look like the following:
+  // {
+  //   name: "assembler",
+  //   nameMatchers: [".asm"],
+  //   singleLineComment: [{ start: ";" }]
+  // }
+  if(typeof(options) !== 'undefined' && typeof(options.pattern) !== 'undefined'){
```

Comment:
Since options is an object, I would use the more readable version (like in line 24)
```js
options && options.pattern
```